### PR TITLE
Update Jenkins core requirement to 2.249.x + Switch tests from Warnings to Warnings Ng Plugin + Update Bill of Materials in integration tests

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Subj. Housekeeping for integration tests and cleanup of some dependency versions

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: integrations/pom.xml [hunk 1]
```diff
@@ -361,10 +364,20 @@
...
 
     <dependency>
-      <groupId>org.jvnet.hudson.plugins</groupId>
-      <artifactId>warnings</artifactId>
-      <version>5.0.2</version>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>warnings-ng</artifactId>
+      <version>8.4.4</version>
       <scope>test</scope>
+      <exclusions>
```
Comment (oleg-nenashev) [inline integrations/pom.xml:371]:
```suggestion
      <exclusions> <!-- TODO: remove once on the 2.263.x baseline -->
```

File: integrations/pom.xml [hunk 2]
```diff
@@ -42,6 +42,12 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>com.github.spotbugs</groupId>
+      <artifactId>spotbugs-annotations</artifactId>
```
Comment (timja) [inline integrations/pom.xml:47]: shouldn't this be coming from jenkins core bom?

https://github.com/jenkinsci/jenkins/blob/master/bom/pom.xml#L54

File: pom.xml [hunk 3]
```diff
@@ -21,7 +21,7 @@
   <properties>
     <revision>1.46</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.222</jenkins.version>
+    <jenkins.version>2.249</jenkins.version>
```
Comment (timja) [inline pom.xml:24]: use the actual LTS version?

```suggestion
    <jenkins.version>2.249.2</jenkins.version>
```

Comment (jetersen) [reply pom.xml:24]: Not sure JCasC is ready to bump to such a recent LTS?

Comment (oleg-nenashev) [reply pom.xml:24]: I would not mind, but I am ready to pause this patch if there is a concern. There were no security releases in 2.249.x , so one can safely stay on 2.235

File: integrations/pom.xml (hunk 2 above)
Comment (oleg-nenashev) [reply integrations/pom.xml:47]: There is an upper bounds conflict with Warnings NG. Will add an exclusion

Comment (oleg-nenashev) [reply integrations/pom.xml:47]: No, I won't. Stapler also has a conflict. I suggest pinning it for now

File: pom.xml (hunk 3 above)
Comment (timja) [reply pom.xml:24]: ah I thought this was integrations, yeah 2.235 might be better

**Comment by oleg-nenashev** [comment]
Let's park it until 2.263.x release then. There are Core BOM updates which
I would like to integrate, but there is no hurry for this PR

File: integrations/pom.xml (hunk 2 above)
Comment (timja) [reply integrations/pom.xml:48]: should be fixed now I think, can you re-check?

File: integrations/pom.xml [hunk 4]
```diff
@@ -367,8 +364,24 @@
       </exclusions>
     </dependency>
 
-    <!-- required to satisfy upper bound dependencies -->
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>warnings-ng</artifactId>
+      <version>8.4.4</version>
```
Comment (timja) [inline integrations/pom.xml:370]: update to latest?

File: integrations/pom.xml [hunk 5]
```diff
@@ -367,8 +364,24 @@
...
+      <scope>test</scope>
+      <exclusions> <!-- TODO: remove once on the 2.263.x baseline -->
+        <exclusion>
+          <groupId>org.slf4j</groupId>
```
Comment (timja) [inline integrations/pom.xml:374]: is this still required?

File: integrations/pom.xml [hunk 6]
```diff
@@ -367,8 +364,24 @@
...
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>commons-io</groupId>
```
Comment (timja) [inline integrations/pom.xml:378]: I expect this isn't required anymore

File: pom.xml (hunk 3 above)
Comment (timja) [reply pom.xml:24]: 2.249 should be fine now, but can we use the actual LTS?

File: integrations/pom.xml (hunk 5 above)
Comment (oleg-nenashev) [reply integrations/pom.xml:374]: Yep. Just resolved the merge conflict

**Pull request closed without merging**